### ISSUE TITLE
Document how to use deployment along with Kafka

### ISSUE THREAD

**Issue by desmarchris** [RasaHQ/rasa-x-helm#207]
Currently, you must add a new section to the values file as well as edit a config map to enable Kafka. This is shown here: https://github.com/RasaHQ/rasa-x-helm/pull/163/files

**Definition of Done**
As a user I can view documentation on how to enable Kafka using helm values.

**Comment by b-quachtran** [comment]
@RasaHQ/infrastructure-squad Are there any plans for supporting this out of the box in the near future?

**Comment by melindaloubser1** [comment]
You can add additional endpoints to `rasa.additionalEndpoints`, so it's not necessary to modify the config map. I just realized this after opening the PR above 😅

**Comment by tczekajlo** [comment]
> @RasaHQ/infrastructure-squad Are there any plans for supporting this out of the box in the near future?

Not really, I mean you can already do this with the rasa helm chart as @melindaloubser1 mentioned. 

For the rasa-x-helm chart, we won't add it, the goal that we have is to separate rasa-x from rasa oss.

**Comment by melindaloubser1** [comment]
I'd suggest we close this since it's already supported/possible at least.

**Comment by desmarchris** [comment]
More of my customers use Kafka than Rabbit, so I think it should be a first class citizen. We shouldn't have to send them examples on how to enable Kafka

**Comment by melindaloubser1** [comment]
That's a fair point, but RabbitMQ is a service started by the helm deployment itself, while Kafka is always external. I think (@tczekajlo correct me if I'm wrong) that is why rabbit got its own section, while Kafka endpoints have to be added in a subsection.

**Comment by desmarchris** [comment]
>while Kafka is always external
are you sure? I didn't think this was the case

**Comment by melindaloubser1** [comment]
well now I'm not sure anymore 😅 But I guess since it requires forking the helm charts to make kafka a service as part of the deployment, I'd never seen anyone do it that way.

**Comment by desmarchris** [comment]
I still think there'd be value in adding a simple `kafka` section like we have in OSS endpoints instead of adding them to a generic area

**Comment by melindaloubser1** [comment]
In that case we can re-open the closed PR since that's what it did - @tczekajlo since it's not really a functional change and it seems it would be more intuitive for users, is that fine?

**Comment by tczekajlo** [comment]
Here are my thoughts 😄 :

1. For now, a way to add extra configuration is disabling rabbitmq (`rabbitmq.enabled=false`) and add Kafka section to `rasa.additionalEndpoints` 

```yaml
#values.yaml
rasa:
  additionalEndpoints:
    event_broker:
      type: "kafka"
      # security protocol
      security_protocol: SASL_PLAINTEXT
      # kafka topic
      topic: my_topic
      # kafka host
      url: localhost
      # SASL credentials
      sasl_username: username
      sasl_password: password

rabbitmq:
  enabled: false
```

2. @melindaloubser1 is right, rabbitmq got his own section because it's a component included in the helm chart.
3. Following the helm good practices, adding `kafka` section will imply that a Kafka helm chart is included and used as a sub-chart, and you can configure it through the `kafka` section, which is not true.

@desmarchris As there is a way to configure Kafka via values without editing the helm chart itself, my proposition would be just to document how to enable Kafka, WDYT?

cc @RasaHQ/infrastructure-squad

**Comment by desmarchris** [comment]
ok fine ;) can we add it to the documentation?

**Comment by tmbo** [comment]
I'll update the issue according to the discussion:
* we won't add native support for kafka
* we will add documentation to document how to enable kafka

**Comment by andreoid** [comment]
Open issue has been moved to JIRA: https://rasahq.atlassian.net/browse/INFRA-7
